src/Internal/Calculator.php

```diff
@@ -93,9 +93,13 @@ private static function detect() : Calculator
      * @param string $a    The first operand.
      * @param string $b    The second operand.
      * @param string $aDig A variable to store the digits of the first operand.
+     * @param-out string
```

Comment:
These `@param-out` annotations are unnecessary for Psalm (and also invalid as they lack a param name – Psalm emits `InvalidDocblock` issues for that).

Psalm assumes that any by-reference params retain their type when passed into a function. If that function _actually_ changes the type (e.g. here: https://psalm.dev/r/983088573a) then you need a `@param-out` annotation. But here, since the type doesn’t change, it's fine.

Reply:
Thanks! I don't know why, but I don't get `InvalidDocblock` locally...

Anyway I changed the code to properly use `@param-out`. The out type is different from the input here, because in the input `null` is allowed.

An example of the error: https://psalm.dev/r/07a893b434
Properly annotated: https://psalm.dev/r/59893b4e3e

@BenMorel, I understand that `string|null` might be a bit confusing, but it's the correct type annotation here. I can do
```php
/**
 * @param string $s
 * @psalm-param string|null $s
 * @param-out string $s
 */
```
but I really don't want to.

Reply:
Actually you can use `@param null` here: https://psalm.dev/r/6d0dfb8a57
Together with `@param-out string $s`

Reply:
True. Changed the PR to `@param null`, please review :)